mmv1/third_party/terraform/tests/resource_cloud_run_service_secret_environment_variable_test.go.erb

```diff
@@ -0,0 +1,118 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' %>
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccCloudRunService_secretEnvironmentVariable(t *testing.T) {
+	t.Parallel()
+
+	project := getTestProjectFromEnv()
+	name := "tftest-cloudrun-" + randString(t, 6)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudRunService_cloudRunServiceUpdateWithSecretEnvVar(name, project, "secret-"+randString(t, 5)),
+			},
+			{
+				ResourceName:            "google_cloud_run_service.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"metadata.0.resource_version", "status.0.conditions"},
+			},
+			{
+				Config: testAccCloudRunService_cloudRunServiceUpdateWithSecretEnvVar(name, project, "secret-"+randString(t, 10)),
+			},
+			{
+				ResourceName:            "google_cloud_run_service.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"metadata.0.resource_version", "status.0.conditions"},
+			},
+		},
+	})
+}
+
+func testAccCloudRunService_cloudRunServiceUpdateWithSecretEnvVar(name, project, secretName string) string {
+	return fmt.Sprintf(`
+data "google_project" "project" {
+}
+
+resource "google_project_service" "secret-manager" {
+  service = "secretmanager.googleapis.com"
+}
+
+resource "google_secret_manager_secret" "secret" {
+  secret_id = "%s"
```

Comment:
Updating the secret_id forces replacement of the secret. This will likely trigger the cloud run service to destroy/recreate rather than update. What we probably want during this update test is to set up 2 different secrets during the first step, then just swap the reference during the second step, only causing an update to the cloud run service

Reply:
Done.